esyi/OpamManifest.ml

```diff
@@ -129,72 +126,57 @@ let convertOpamFormula f =
 let convertOpamUrl (manifest : t) =
   let open Result.Syntax in
 
+  let convChecksum hash =
+    match OpamHash.kind hash with
+    | `MD5 -> Checksum.Md5, OpamHash.contents hash
+    | `SHA256 -> Checksum.Sha256, OpamHash.contents hash
+    | `SHA512 -> Checksum.Sha512, OpamHash.contents hash
+  in
+
+  let convUrl (url : OpamUrl.t) =
+    match url.backend with
+    | `http -> return (OpamUrl.to_string url)
+    | _ -> errorf "unsupported dist for opam package: %s" (OpamUrl.to_string url)
+  in
+
   let sourceOfOpamUrl url =
-    let%bind checksum =
-      let checksums = OpamFile.URL.checksum url in
-      let f c =
-        match OpamHash.kind c with
-        | `MD5 -> Checksum.Md5, OpamHash.contents c
-        | `SHA256 -> Checksum.Sha256, OpamHash.contents c
-        | `SHA512 -> Checksum.Sha512, OpamHash.contents c
-      in
-      match List.map ~f checksums with
-      | [] ->
-        let msg =
-          Format.asprintf
-            "no checksum provided for %s@%s"
-            (OpamPackage.Name.to_string manifest.name)
-            (OpamPackage.Version.to_string manifest.version)
-        in
-        Error msg
-      | checksum::_ -> Ok checksum
-    in
 
-    let convert (url : OpamUrl.t) =
-      match url.backend with
-      | `http ->
-        return (Dist.Archive {
-          url = OpamUrl.to_string url;
-          checksum;
-        })
-      | `rsync -> Error "unsupported source for opam: rsync"
-      | `hg -> Error "unsupported source for opam: hg"
-      | `darcs -> Error "unsupported source for opam: darcs"
-      | `git -> Error "unsupported source for opam: git"
+    let%bind hash =
+      match OpamFile.URL.checksum url with
+      | [] ->
+        errorf
+          "no checksum provided for %s@%s"
+          (OpamPackage.Name.to_string manifest.name)
+          (OpamPackage.Version.to_string manifest.version)
+      | hash::_ -> return hash
     in
 
-    let%bind main = convert (OpamFile.URL.url url) in
     let mirrors =
+      let urls =
+        (OpamFile.URL.url url)
+        ::(OpamFile.URL.mirrors url)
+      in
       let f mirrors url =
-        match convert url with
-        | Ok mirror -> mirror::mirrors
+        match convUrl url with
+        | Ok url -> Dist.Archive {url; checksum = convChecksum hash;}::mirrors
         | Error _ -> mirrors
       in
-      List.fold_left ~f ~init:[] (OpamFile.URL.mirrors url)
+      List.fold_left ~f ~init:[] urls
     in
-    return (main, mirrors)
-  in
-
-  let%bind main, mirrors =
-    match manifest.url with
-    | Some url -> sourceOfOpamUrl url
-    | None ->
-      let main = Dist.NoSource in
-      Ok (main, [])
-  in
 
-  match manifest.archive with
-  | Some archive ->
-    let mirrors = main::mirrors in
     let main =
-      Dist.Archive {
-        url = archive.url;
-        checksum = Checksum.Md5, archive.md5;
-      }
+      let url = "https://opam.ocaml.org/cache/" ^ String.concat "/" (OpamHash.to_path hash) in
```

Comment:
We hard code this for now as we fetch opam repo via VCS and it doesn't have this info in `/repo` config. That's ok for now as we officially only support main opam repo.